src/gui/widgets/Knob.cpp

```diff
@@ -618,12 +645,55 @@ void Knob::mousePressEvent( QMouseEvent * _me )
 
 void Knob::mouseMoveEvent( QMouseEvent * _me )
 {
-	if( m_buttonPressed && _me->pos() != m_origMousePos )
+	if( m_buttonPressed && _me->pos() != m_lastMousePos )
 	{
-		m_mouseOffset = _me->pos() - m_origMousePos;
-		setPosition( m_mouseOffset );
+		// knob position is changed depending on last mouse position
+		setPosition( _me->pos() - m_lastMousePos );
 		emit sliderMoved( model()->value() );
-		QCursor::setPos( mapToGlobal( m_origMousePos ) );
+
+		// the rest of the code updates the cursor and/or m_origMousePos
+
+		// get screen min/max values where a flip is not required yet
+		const QRect& rec =
+#if QT_VERSION >= 0x050A00
+			QApplication::screenAt(QCursor::pos())->geometry();
+#else
+			QApplication::desktop()->availableGeometry();
+#endif
+		int ymax = qMax(rec.height() - 2, 0), xmax = qMax(rec.width() - 2, 0);
+		int ymin = 1, xmin = 1;
+
+		// calculate new coordinates
+		int x = QCursor::pos().x(), y = QCursor::pos().y();
+		QPoint newPos(
+			(x<xmin) ? xmax : (x>xmax) ? xmin : x,
+			(y<ymin) ? ymax : (y>ymax) ? ymin : y);
+
+		auto setCursor = [&](const QPoint& p) -> bool
```

Comment:
Can you use `s_setPosStatus` or cache the checked result, if that works?

Reply:
I [tried](https://github.com/JohannesLorenz/lmms/pull/4/commits/33dd543f7f07f667efc29a3fa8e4c8ecfcc39ed5), but somehow this causes the knob resetting to the opposite site even on native Arch Linux where `setPos` works.